playbooks/bond.yml

```diff
@@ -0,0 +1,24 @@
+---
+- hosts: all
+  become: True
+  tasks:
+    - nmstate_apply:
+        state:
+          interfaces:
+            - name: bond99
+              type: bond
+              state: up
+              ipv4:
+                address:
+                  - ip: 10.10.10.10
+                    prefix-length: 24
+                enabled: true
+              link-aggregation:
+                mode: balance-rr
+                options:
+                  miimon: '140'
+                slaves:
```

Comment:
Not sure, but would it be possible to use `port` instead? Thanks. @tyll

Reply:
It is possible if Nmstate supports it. The Ansible module just relays the state to libnmstate.
@liangwen12year please also update the nmstate.io documentation to use port.

Reply:
Yes, since nmstate 0.4 it should work :-)

Reply:
I failed in using `port`, error attached```fatal: [cen]: FAILED! => {"changed": false, "module_stderr": "Shared connection to cen closed.\r\n", "module_stdout": "\r\nTraceback (most recent call last):\r\n  File \"/home/osboxes/.ansible/tmp/ansible-tmp-1603301308.2269027-48796-239240355366266/AnsiballZ_nmstate_apply.py\", line 102, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/osboxes/.ansible/tmp/ansible-tmp-1603301308.2269027-48796-239240355366266/AnsiballZ_nmstate_apply.py\", line 94, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/osboxes/.ansible/tmp/ansible-tmp-1603301308.2269027-48796-239240355366266/AnsiballZ_nmstate_apply.py\", line 40, in invoke_module\r\n    runpy.run_module(mod_name='ansible.modules.nmstate_apply', init_globals=None, run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\r\n    mod_name, mod_spec, pkg_name, script_name)\r\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_nmstate_apply_payload_uurk98xb/ansible_nmstate_apply_payload.zip/ansible/modules/nmstate_apply.py\", line 110, in <module>\r\n  File \"/tmp/ansible_nmstate_apply_payload_uurk98xb/ansible_nmstate_apply_payload.zip/ansible/modules/nmstate_apply.py\", line 106, in main\r\n  File \"/tmp/ansible_nmstate_apply_payload_uurk98xb/ansible_nmstate_apply_payload.zip/ansible/modules/nmstate_apply.py\", line 102, in run_module\r\n  File \"/tmp/ansible_nmstate_apply_payload_uurk98xb/ansible_nmstate_apply_payload.zip/ansible/modules/nmstate_apply.py\", line 79, in run\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/deprecation.py\", line 40, in wrapper\r\n    return func(*args, **kwargs)\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/nm/nmclient.py\", line 96, in wrapped\r\n    ret = func(*args, **kwargs)\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/netapplier.py\", line 73, in apply\r\n    state.State(desired_state), verify_change, commit, rollback_timeout\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/netapplier.py\", line 175, in _apply_ifaces_state\r\n    _verify_change(desired_state)\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/netapplier.py\", line 221, in _verify_change\r\n    verifiable_desired_state.verify_interfaces(current_state)\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/state.py\", line 330, in verify_interfaces\r\n    self._assert_interfaces_equal(other_state)\r\n  File \"/usr/lib/python3.6/site-packages/libnmstate/state.py\", line 759, in _assert_interfaces_equal\r\n    current_state.interfaces[ifname],\r\nlibnmstate.error.NmstateVerificationError: \r\ndesired\r\n=======\r\n---\r\nname: bond99\r\ntype: bond\r\nstate: up\r\nipv4:\r\n  address:\r\n  - ip: 10.10.10.15\r\n    prefix-length: 24\r\n  dhcp: false\r\n  enabled: true\r\nipv6:\r\n  enabled: false\r\nlink-aggregation:\r\n  mode: balance-rr\r\n  options:\r\n    ad_actor_sys_prio: ''\r\n    ad_user_port_key: ''\r\n    miimon: 140\r\n  ports:\r\n  - name: eth1\r\n  - name: eth1.ep\r\n  slaves:\r\n  - eth1\r\n  - eth1.ep\r\nmac-address: 08:00:27:87:43:C3\r\nmtu: 1500\r\n\r\ncurrent\r\n=======\r\n---\r\nname: bond99\r\ntype: bond\r\nstate: up\r\nipv4:\r\n  address:\r\n  - ip: 10.10.10.15\r\n    prefix-length: 24\r\n  dhcp: false\r\n  enabled: true\r\nipv6:\r\n  enabled: false\r\nlink-aggregation:\r\n  mode: balance-rr\r\n  options:\r\n    ad_actor_sys_prio: ''\r\n    ad_user_port_key: ''\r\n    miimon: 140\r\n  slaves:\r\n  - eth1\r\n  - eth1.ep\r\nmac-address: 08:00:27:87:43:C3\r\nmtu: 1500\r\n\r\ndifference\r\n==========\r\n--- desired\r\n+++ current\r\n@@ -16,9 +16,6 @@\r\n     ad_actor_sys_prio: ''\r\n     ad_user_port_key: ''\r\n     miimon: 140\r\n-  ports:\r\n-  - name: eth1\r\n-  - name: eth1.ep\r\n   slaves:\r\n   - eth1\r\n   - eth1.ep\r\n\r\n\r\n", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}```

Reply:
No worries I will take a look later to this failure. Let's keep slaves for now.